Sources/GRInputField/SwiftUI/InputField.swift

```diff
@@ -0,0 +1,279 @@
+        case .valid:
+            uiView.unfail()
+
+        case .invalid(let error):
+            uiView.failSilently(with: error.localizedDescription)
+
+        case .none:
+            setValidityStateInGroup(to: .pending(uiView.validateSilently()), in: context)
+
+        case .pending:
+            break
+        }
+    }
+
+    private func setValidityStateInGroup(to validationState: ValidationState, in context: Context) {
+        Task { @MainActor in
+            validityGroup.updateValue(
+                validationState,
+                forKey: context.coordinator.textFieldUUID
+            )
+        }
+    }
+
+    private func invalidateValidityState(in context: Context) {
+        Task { @MainActor in
+            validityGroup.removeValue(forKey: context.coordinator.textFieldUUID)
+        }
+    }
+
+}
+
+// MARK: - Public modifiers
+
+public extension InputField {
+
+    @available(iOS 15.0, *)
+    func bindFocusState<Focus: RawRepresentable & CaseIterable & Hashable>(
+        _ focusState: FocusState<Focus?>.Binding,
+        to field: Focus
+    ) -> some View where Focus.RawValue == Int {
+        var modifiedSelf = self
+        modifiedSelf.returnAction = { focusNextField(in: focusState) }
+        return modifiedSelf.focused(focusState, equals: field)
+    }
+
+    func validityGroup(_ binding: Binding<ValidityGroup>) -> Self {
+        var modifiedSelf = self
+        modifiedSelf._validityGroup = binding
+        return modifiedSelf
+    }
+
+    func onResign(_ action: @escaping VoidClosure) -> Self {
+        var modifiedSelf = self
+        modifiedSelf.resignAction = action
+        return modifiedSelf
+    }
+
+    func validationCriteria(@ValidatorBuilder _ criteria: @escaping () -> Validator) -> Self {
+        var modifiedSelf = self
+        modifiedSelf.criteria = criteria
+        return modifiedSelf
+    }
+
+    func inputFieldTraits(
+        textContentType: UITextContentType? = .none,
+        autocapitalizationType: UITextAutocapitalizationType = .none,
+        autocorrectionType: UITextAutocorrectionType = .default,
+        keyboardType: UIKeyboardType = .default,
+        returnKeyType: UIReturnKeyType = .default,
+        numpadReturnKeyTitle: String? = "Done",
+        clearButtonMode: UITextField.ViewMode = .whileEditing,
+        isSecureTextEntry: Bool = false,
+        isHapticsAllowed: Bool = true
+    ) -> Self {
+        var modifiedSelf = self
+        modifiedSelf.traits = InputFieldTraits(
+            textContentType: textContentType,
+            autocapitalizationType: autocapitalizationType,
+            autocorrectionType: autocorrectionType,
+            keyboardType: keyboardType,
+            returnKeyType: returnKeyType,
+            numpadReturnKeyTitle: numpadReturnKeyTitle,
+            clearButtonMode: clearButtonMode,
+            isSecureTextEntry: isSecureTextEntry,
+            isHapticsAllowed: isHapticsAllowed
+        )
+        return modifiedSelf
+    }
+
+}
```

Comment:
Nebolo by lepsie toto spravit cez `@Environment`? Dalo by sa tym padom applikovat modifer napr. na celu groupu textfieldov. Taktiez by nezalezalo na poradi modiferov. Teraz staci za napises takyto kod:
<img width="988" alt="Snímka obrazovky 2024-06-28 o 9 17 03" src="https://github.com/GoodRequest/GoodSwiftUI/assets/114930016/4e55b912-8855-4cda-a9f9-b999e643eba1">

a uz to nebude fungovat. Pretoze vsetky ostatne viewcka ty vratia some View a nie ten InputField

Reply:
budem robit do Swift 6 verzie